src/sentry/static/sentry/app/components/organizations/multipleEnvironmentSelector.jsx

```diff
@@ -57,6 +58,25 @@ class MultipleEnvironmentSelector extends React.PureComponent {
     };
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    // When projects change we may need to update the selected value if the current values
+    // become invalid.
+    const sharedProjects = intersection(
+      prevProps.selectedProjects,
+      this.props.selectedProjects
+    );
+    if (sharedProjects.length !== prevProps.selectedProjects.length) {
+      const selectedEnvs = Array.from(this.state.selectedEnvs.values());
```

Comment:
Should we make sure `this.state.selectedEnvs` gets updated as well?

Reply:
Good point! I can do that.